File: game/cave/waiting_input.py
import pygame

from cave.cave_resources import CaveResources
from cave.going_rope import GoingRope
from config import Config
from phone_events import PhoneEvents
from state import State


class WaitingInput(State):
    def process_events(self, phone_events: PhoneEvents):
        if phone_events.press_3:
            self.context.cave_selected_rope = 0
            return GoingRope
        elif phone_events.press_6:
            self.context.cave_selected_rope = 1
            return GoingRope
        elif phone_events.press_9:
            self.context.cave_selected_rope = 2
            return GoingRope

        if self.one_shot(0.5, "afskylia_snak"):
            pygame.mixer.Sound.play(CaveResources.afskylia_snak)

        return None

    def render(self, screen):
        screen.blit(CaveResources.first_rope[0], (0, 0))

    def on_enter(self) -> None:
        super().on_enter()
        if not Config.ARGENTINE_VERSION:
            pygame.mixer.Sound.play(CaveResources.stemning)

    def on_exit(self) -> None:
        super().on_exit()
        pygame.mixer.Sound.play(CaveResources.tast_trykket)

File: game-web/src/cave/waitingInput.ts
/** Port of game/cave/waiting_input.py */

import { State, type StateClass } from '../state';
import { AudioHelper } from '../audioHelper';
import { CaveResources } from './caveResources';
import { GoingRope } from './goingRope';
import { Config } from '../config';
import type { PhoneEvents } from '../phoneEvents';

export class WaitingInput extends State {
  processEvents(phoneEvents: PhoneEvents): StateClass | null {
    if (phoneEvents.press_3) { this.context.caveSelectedRope = 0; return GoingRope; }
    if (phoneEvents.press_6) { this.context.caveSelectedRope = 1; return GoingRope; }
    if (phoneEvents.press_9) { this.context.caveSelectedRope = 2; return GoingRope; }
    if (this.oneShot(0.5, 'afskylia_snak')) AudioHelper.play(CaveResources.afskyliaSnak);
    return null;
  }

  onEnter(): void {
    super.onEnter();
    if (!Config.ARGENTINE_VERSION) {
      this.context.caveStemningId = AudioHelper.play(CaveResources.stemning);
    }
  }

  onExit(): void {
    super.onExit();
    AudioHelper.play(CaveResources.tastTrykket);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(CaveResources.firstRope[0], 0, 0);
  }
}
